In [1]:
from pathlib import Path
from keyword_search_evaluation import get_scores

/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
MODELS_PATH = Path("models")

In [3]:
def print_model_scores(model_path: str | Path):
    scores = get_scores(model_path)
    print(model_path)
    print(f"Full match score: {scores["perfect_match_score"] * 100:.2f}%")
    print(f"Related keywords score: {scores["related_keyword_score"] * 100:.2f}%")

In [4]:
for model_path in MODELS_PATH.iterdir():
    print_model_scores(model_path)
    print("")

You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.
Loading weights: 100%|██████████| 199/199 [00:00<00:00, 833.26it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/vanilla-finetuned
Full match score: 43.68%
Related keywords score: 5.86%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 905.89it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-0.50
Full match score: 46.05%
Related keywords score: 6.05%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 902.15it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-0.75
Full match score: 45.55%
Related keywords score: 6.06%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 903.06it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-1.00
Full match score: 45.95%
Related keywords score: 6.12%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 875.99it/s, Materializing param=pooler.dense.weight]                               


models/jargon-layman-finetuned-0.25
Full match score: 45.65%
Related keywords score: 5.99%



In [5]:
print_model_scores("allenai/scibert_scivocab_uncased")

No sentence-transformers model found with name allenai/scibert_scivocab_uncased. Creating a new one with mean pooling.
Loading weights: 100%|██████████| 199/199 [00:00<00:00, 1185.55it/s, Materializing param=pooler.dense.weight]                               
BertModel LOAD REPORT from: allenai/scibert_scivocab_uncased
Key                                        | Status     |  | 
-------------------------------------------+------------+--+-
cls.predictions.transform.LayerNorm.bias   | UNEXPECTED |  | 
cls.predictions.decoder.weight             | UNEXPECTED |  | 
cls.predictions.decoder.bias               | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.weight | UNEXPECTED |  | 
cls.predictions.transform.dense.bias       | UNEXPECTED |  | 
cls.predictions.transform.dense.weight     | UNEXPECTED |  | 
cls.seq_relationship.weight                | UNEXPECTED |  | 
cls.predictions.bias                       | UNEXPECTED |  | 
cls.seq_relationship.bias                  | UNEXPECTED |  

allenai/scibert_scivocab_uncased
Full match score: 11.46%
Related keywords score: 1.94%


# Results

| Model | Exact keyword match score | Related keyword score |
|---|---|---|
| allenai/scibert_scivocab_uncased | 11.46% | 1.94% |
| vanilla-finetuned | <u>43.68%</u> | <u>5.86%</u> |
| jargon-layman-finetuned | **45.95%** | **6.12%** |

In [6]:
import pandas as pd

In [7]:
scores_for_select_model = get_scores("models/jargon-layman-finetuned-1.00")
search_results: list[list[dict[str, float]]] = scores_for_select_model["search_results"]
search_df: pd.DataFrame = scores_for_select_model["search_df"]

You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.
Loading weights: 100%|██████████| 199/199 [00:00<00:00, 975.47it/s, Materializing param=pooler.dense.weight]                               


In [8]:
search_results

[[{'corpus_id': 0, 'score': 0.9999999403953552},
  {'corpus_id': 861, 'score': 0.8305516242980957},
  {'corpus_id': 597, 'score': 0.7182247638702393},
  {'corpus_id': 11, 'score': 0.7039348483085632},
  {'corpus_id': 1, 'score': 0.7026268839836121},
  {'corpus_id': 595, 'score': 0.6995381712913513},
  {'corpus_id': 599, 'score': 0.6906185150146484},
  {'corpus_id': 855, 'score': 0.6670401096343994},
  {'corpus_id': 14, 'score': 0.6629663109779358},
  {'corpus_id': 858, 'score': 0.6508963108062744},
  {'corpus_id': 10, 'score': 0.6384038925170898},
  {'corpus_id': 600, 'score': 0.6352406144142151},
  {'corpus_id': 849, 'score': 0.6288851499557495},
  {'corpus_id': 732, 'score': 0.6273906230926514},
  {'corpus_id': 6, 'score': 0.6255139112472534}],
 [{'corpus_id': 0, 'score': 0.6528292894363403},
  {'corpus_id': 1, 'score': 0.635375440120697},
  {'corpus_id': 171, 'score': 0.6244080662727356},
  {'corpus_id': 173, 'score': 0.6028590202331543},
  {'corpus_id': 11, 'score': 0.5801800489425

In [9]:
search_df

,path,jargon,layman
0,data/test_keywords/0a5325adb8861db65ff96c3516b...,Art installation,Art installation
1,data/test_keywords/0a5325adb8861db65ff96c3516b...,Composting area,Place where waste breaks down
2,data/test_keywords/0a5325adb8861db65ff96c3516b...,Pollination,Flowers being fertilized by insects
3,data/test_keywords/0a5325adb8861db65ff96c3516b...,Subject of sense experience,Person having a sensory experience
4,data/test_keywords/0a5325adb8861db65ff96c3516b...,Material processes,Natural processes in the environment
...,...,...,...
1007,data/test_keywords/06aef5ffb118088f7522be1b64e...,Finite number of flips,Polygon becomes convex in a limited number of ...
1008,data/test_keywords/06aef5ffb118088f7522be1b64e...,Self-crossing polygons,Polygons that intersect themselves
1009,data/test_keywords/06aef5ffb118088f7522be1b64e...,Θ(n²) time,Time grows with square of number of sides
1010,data/test_keywords/06aef5ffb118088f7522be1b64e...,O(n) time,Time grows linearly with number of sides


In [10]:
results_df = pd.DataFrame(data={
    "query_index": range(len(search_results)),
    "result_id": [
        [result["corpus_id"] for result in single_results]
        for single_results in search_results
    ]
}).explode("result_id")
results_df.head()

,query_index,result_id
0,0,0
0,0,861
0,0,597
0,0,11
0,0,1


In [11]:
results_df["layman"] = results_df["query_index"].apply(lambda i: search_df["layman"].iloc[i])
results_df["jargon"] = results_df["result_id"].apply(lambda i: search_df["jargon"].iloc[i])
results_df.head()

,query_index,result_id,layman,jargon
0,0,0,Art installation,Art installation
0,0,861,Art installation,Art exhibition
0,0,597,Art installation,American art
0,0,11,Art installation,Participatory regime of art
0,0,1,Art installation,Composting area


In [12]:
search_results[0][0]

{'corpus_id': 0, 'score': 0.9999999403953552}